paasta_tools/utils.py

```diff
@@ -2802,6 +2814,17 @@ def get_eks_cluster_aliases(self) -> Dict[str, str]:
     def get_cluster_pools(self) -> Dict[str, List[str]]:
         return self.config_dict.get("allowed_pools", {})
 
+    def get_spark_driver_iam_role(self) -> str:
+        return self.get_spark_run_config().get("default_spark_driver_iam_role", "")
+
+    def get_spark_executor_iam_role(self) -> str:
+        # use the same IAM role as the Spark driver
+        return self.get_spark_run_config().get("default_spark_driver_iam_role", "")
+
```

Comment:
I don't have a ton of context on this change, but from a security standpoint I am pretty strongly against the concept of a "default executor role", because it steers people away from being intentional about the permissions their spark jobs need, and I fear people will just use the default role and over time the default role will have a bunch of permissions that become hard to reason about ensure least-privilege best practices.

Reply:
@danielpops the default executor role is the one managed by ML Compute with the least privileges required to run a Spark job. We have no intention of adding extra permissions to this role and if used, it will fail the jobs which require additional permissions.

Reply:
That's great to hear! I'm not sure how we'd prevent people from presuming the solution to fix a failed job is to add more permissions to the spark driver role, but as long as you're committed to preventing that from happening, that sounds good to me :)

Reply:
That's great to hear! I'm not sure how we'd prevent people from presuming the solution to fix a failed job is to add more permissions to the spark driver role, but as long as you're committed to preventing that from happening, that sounds good to me :)